### PR TITLE
[MID-28] add ClientPortal and Modal components

### PR DESCRIPTION
**Pull request by Mykhailo-Pasternak**
Ticket: https://trello.com/c/s1R48pYt/28-%D0%B7%D0%B2%D0%BE%D1%80%D0%BE%D1%82%D0%BD%D1%96%D0%B9-%D0%B7%D0%B2%D1%8F%D0%B7%D0%BE%D0%BA-%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82-%D0%BC%D0%BE%D0%B4%D0%B0%D0%BB%D0%BA%D0%B0

**A modal window is created using the ClientPortal.** 
The modal window has basic functionality, including the ability to open and close the window both by pressing a button and clicking outside the internal component boundaries.

### Look here, the deployed version:
https://mayak-nb9v55rav-pozakhmarnyi.vercel.app/

### CODE REVIEW

File: src/app/_components/Header/Header.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+  const toggleModal = () => {
+    setIsModalOpen(prevState => !prevState);
+  };
   return (
-    <header className="bg-primary-100 p-5 text-h1 text-primary-700">
-      Header
+    <header className="flex justify-between bg-primary-100">
+      <h1 className="p-3 text-h1 text-primary-700"> Header </h1>
+      <button
+        onClick={() => toggleModal()}
```
Comment (steven-tailor) [inline src/app/_components/Header/Header.js]: No reason to wrap callback
```suggestion
        onClick={toggleModal}
```

File: src/app/_components/Header/Header.js [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+        className="rounded-lg text-primary-500 outline"
+      >
+        Зворотній зв`язок
+      </button>
+
+      <Modal isOpen={isModalOpen} onClose={toggleModal}>
+        <div className="rounded-md bg-primary-300 p-4 shadow-md">
+          <h2>This is a Modal</h2>
+          <p>Modal content goes here.</p>
```
Comment (steven-tailor) [inline src/app/_components/Header/Header.js]: We can't merge it as is. I guess that should be passed as a prop or child

File: src/app/_components/Header/Header.js [hunk 3]
```diff
@@ -1,9 +1,29 @@
-import React from 'react';
+'use client';
+
+import React, { useState } from 'react';
+import Modal from '../Modal';
 
 export default function Header() {
```
Comment (steven-tailor) [inline src/app/_components/Header/Header.js]: NIT(Just advice): default exports are bad practice in general, cuz can let you `import Foo from 'bar'` and vice-versa. Meanwhile named export will prevent such cases (`import {Foo} from 'foo'`). Try to use default exports only if that's really necessary.

**Review by steven-tailor — CHANGES_REQUESTED** [review]

**Comment by steven-tailor** [comment]
@Mykhailo-Pasternak Have you checked if there are no unwanted renders/flickering of the modal (using dev tools)?

**Comment by Mykhailo-Pasternak** [comment]
Expanded the versatility of modal window usage (with/without default 'X' button and background blur)

File: src/app/_components/Header/Header.js [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
+        Зворотній зв`язок
+      </button>
+
+      <Modal
+        isOpen={isModalOpen}
+        onClose={toggleModal}
+        isBlurBackground={false}
+      >
+        {Example}
+      </Modal>
```
Comment (alex-popov-tech) [inline src/app/_components/Header/Header.js]: 🟠 i think there is no need to put example modal into commits and codebase, you can add it to PR description ( and deploy live example from local to see it in action )

File: src/app/_components/Modal/Modal.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <ClientPortal selector="modal-root" show={isOpen}>
+      {isOpen && (
+        <>
+          {isBlurBackground && blurBackground}
+          <div
+            className="fixed left-0 top-0 flex h-full w-full items-center justify-center"
+            onClick={onClose}
+          >
+            {isCloseButton ? (
+              <DefaultModalCloseButton onClose={onClose}>
```
Comment (alex-popov-tech) [inline src/app/_components/Modal/Modal.js]: 🟠 looks a bit weird when you are passing modal content ( clihdren ) into close button component, is that intended?

File: src/app/_components/Header/index.js [hunk 6]
```diff
@@ -1 +1 @@
-export { default } from './Header';
+export { Header } from './Header';
```
Comment (alex-popov-tech) [inline src/app/_components/Header/index.js]: 🟠 aren't we using default imports for components? can we stick to named/default everywhere? ( i prefer named )
🟢 can `Header` folder be converted to just `Header.js` file with component?

**Review by alex-popov-tech — CHANGES_REQUESTED** [review]

File: src/app/_components/Header/index.js (hunk 6 above)
Comment (Mykhailo-Pasternak) [reply src/app/_components/Header/index.js]: image

File: src/app/_components/Modal/DefaultModalCloseButton.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    onClick={(e) => {
+      e.stopPropagation();
+    }}
+  >
+    {children}
+    <button
+      onClick={onClose}
+      className="absolute right-4 top-4 hover:text-primary-500"
+    >
+      <svg
```
Comment (steven-tailor) [inline src/app/_components/Modal/DefaultModalCloseButton.js]: let's store SVG as a separate file - it creates a lot of mess.

File: src/app/_components/Header/index.js (hunk 6 above)
Comment (Mykhailo-Pasternak) [reply src/app/_components/Header/index.js]: 🟢If you don't use the index.js file:
then the imports will look like this:
image
and with them like this:
image

File: src/app/_components/Modal/Modal.js (hunk 5 above)
Comment (Mykhailo-Pasternak) [reply src/app/_components/Modal/Modal.js]: ok, changed to: ModalCloseButton

File: src/app/_components/Header.js [hunk 8]
```diff
@@ -1,15 +1,23 @@
-import React from 'react';
+'use client';
```
Comment (alex-popov-tech) [inline src/app/_components/Header.js]: 🟠 can we maybe separate this button into component, to not make whole header client-side?

File: src/app/_components/Modal/ModalCloseButton.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import PropTypes from 'prop-types';
+import СloseIcon from '../../../../public/assets/icons/closeIcon.svg';
+
+export const ModalCloseButton = ({ onClose }) => (
+  <button onClick={onClose} className="p-[9.3px]">
+    <СloseIcon
+      alt="Сlose Icon"
+      aria-label="Сlose Icon"
+      priority="true"
+      className={'text-gray-700 hover:text-primary-400'}
```
Comment (alex-popov-tech) [inline src/app/_components/Modal/ModalCloseButton.js]: 🟢 i think you don't need `{}` here

File: src/app/_components/Header.js (hunk 8 above)
Comment (Mykhailo-Pasternak) [reply src/app/_components/Header.js]: Of course. In fact, whoever adds that feedback form will also use this modal window.

At the moment, it doesn't make sense for me to create some kind of component that will respond to the modal window's enable handler.

'use client' - was added to demonstrate / be able to use the modal window.
Now I will clean all the excess.

**Review by alex-popov-tech — APPROVED** [review]

**Review by watchlar2000 — APPROVED** [review]

**Review by steven-tailor — APPROVED** [review]

**Pull request merged**